### ISSUE TITLE
Lossy compression of screen?

### ISSUE THREAD

**Issue by monken** [neutrinolabs/xrdp#817]
It seems like currently, only lossless compression is supported. I have set up a Linux VM and connect through XRDP from a Windows machine. In parallel, I also use Guacamole (guac-dev.org) to connect to the same host. I usually get better performance out of Guacamole when I'm on a high-latency link (cross atlantic). I attribute this to the lossy compression that Guacamole is using as opposed to the lossless compression from xrdp. Is it possible to tweak the compression algorithm to reduce bandwidth and therefore improve performance over high-latency/low-bandwidth links?

**Comment by johnarnold** [comment]
Are you using latest version of Xorg driver (0.2.2)  ?  Recently there were big improvements with tile-based refresh that really improved performance for me over WAN links, VPNs etc.

**Comment by monken** [comment]
Yes, that update definitely helped. And I also would like to mention that I prefer the lossless compression most days out of the week. But the few time where I am on a slow link, it becomes almost impossible to work with the remote desktop.

**Comment by metalefty** [comment]
Need to implement RemoteFX Adaptive Graphics or Progressive RemoteFX 
Maybe they're part of MS-RDPEGFX: https://msdn.microsoft.com/en-us/library/hh554846.aspx

**Comment by shadoxx** [comment]
+1 for this feature request. Implementing the MS-RDPEGFX specification into XRDP would be a huge step forward. Microsoft just updated the specification today: https://winprotocoldoc.blob.core.windows.net/productionwindowsarchives/MS-RDPEGFX/[MS-RDPEGFX]-170915-diff.pdf

**Comment by matt335672** [comment]
Over a WAN, see also #2910. This doesn't implement lossy compression, but it does improve throughput.

**Comment by matt335672** [comment]
Just found this one - we now support H.264 so this one can be closed.
